src/test/java/org/datadog/jmxfetch/TestReconnectContainer.java

```diff
@@ -0,0 +1,200 @@
+package org.datadog.jmxfetch;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.management.MBeanServerConnection;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.DockerImageName;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.datadog.jmxfetch.reporter.ConsoleReporter;
+
+@Slf4j
+class JMXFailServerClient {
```

Comment:
this class would likely make sense to live in a separate file, maybe in `TestRemoteAppCommon`?

(Also rename from `JMXFailServerClient` to something ... better)